File: src/api/tray.ts
import { invokeCommand } from './client';

export interface TrayMenuText {
  showText: string;
  settingsText: string;
  quitText: string;
}

export async function updateTrayMenu(text: TrayMenuText): Promise<void> {
  await invokeCommand<void>('update_tray_menu', {
    showText: text.showText,
    settingsText: text.settingsText,
    quitText: text.quitText,
  });
}

File: src-tauri/src/commands/tray.rs
use crate::error::AppError;
use crate::services::tray_service;
use tracing::instrument;

#[tauri::command]
#[instrument]
pub fn update_tray_menu(
    app: tauri::AppHandle,
    show_text: String,
    settings_text: String,
    quit_text: String,
) -> Result<(), AppError> {
    tray_service::update_tray_menu(&app, &show_text, &settings_text, &quit_text)
}
